### ISSUE TITLE
Rustfmt fails on long trait bounds 

### ISSUE THREAD

**Issue by phil-opp** [rust-lang-nursery/rustfmt#1168]
Example: https://is.gd/ggx0k5

Before rustfmt:

``` rust
pub trait Number: Copy + Eq +      Not<Output = Self> + Shl<u8, Output = Self> +
                        Shr<u8, Output = Self>       +
    BitAnd<Self, Output=Self> +    BitOr<Self, Output=Self>  + BitAndAssign + BitOrAssign



    {
    // test
    fn zero() -> Self;
}

fn main() {}
```

After rustfmt:

``` rust
pub trait Number: Copy + Eq +      Not<Output = Self> + Shl<u8, Output = Self> +
                        Shr<u8, Output = Self>       +
    BitAnd<Self, Output=Self> +    BitOr<Self, Output=Self>  + BitAndAssign + BitOrAssign



    {
// test
    fn zero() -> Self;
}

fn main() {}
```

So rustfmt doesn't fix this formatting at all. It only removes the correct indentation of the `// test` comment.

**Comment by bjgill** [comment]
So I've had a quick investigation, and it looks like the root cause is that in `items::format_trait()`, the calculation of `where_budget` uses `try_opt!()`. I suspect this is a mistake, but I haven't yet worked out what a correct replacement would be.

**Comment by clarcharr** [comment]
This is still relevant in 0.8; is there a fmt RFC for how to handle long trait bounds?

**Comment by topecongiro** [comment]
Closed via #1647.
